straxen/plugins/acqmon_processing.py

```diff
@@ -13,35 +17,108 @@
+    """
+    save_when = strax.SaveWhen.TARGET
+    __version__ = '1.0.0'
+    hit_min_amplitude_aqmon = straxen.URLConfig(
+        default=(
+            # Analogue signals
+            (50, (int(AqmonChannels.SUM_WF),)),
+            # Digital signals, can set a much higher threshold
+            (1500, (
+                int(AqmonChannels.GPS_SYNC),
+                int(AqmonChannels.M_VETO_SYNC),
+                int(AqmonChannels.HEV_STOP),
+                int(AqmonChannels.HEV_START),
+                int(AqmonChannels.HE_STOP),
+                int(AqmonChannels.HE_START),
+                int(AqmonChannels.BUSY_STOP),
+                int(AqmonChannels.BUSY_START),)),
+            # Fake signals, 0 meaning that we won't find hits using
+            # strax but just look for starts and stops
+            (0, (int(AqmonChannels.ARTIFICIAL_DEADTIME),)),
+
+        ),
+        track=True,
+        help='Minimum hit threshold in ADC*counts above baseline. Specified '
+             'per channel in the format (threshold, (chx,chy),)',
+    )
+    baseline_samples_aqmon = straxen.URLConfig(
+        default=10,
+        track=True,
+        help='Number of samples to use at the start of the pulse to determine the baseline'
+    )
+
+    depends_on = 'raw_records_aqmon'
+    provides = 'aqmon_hits'
+    data_kind = 'aqmon_hits'
 
     dtype = strax.hit_dtype
 
-    save_when = strax.SaveWhen.TARGET
-
     def compute(self, raw_records_aqmon):
-        rec = strax.raw_to_records(raw_records_aqmon)
-        strax.sort_by_time(rec)
-        strax.zero_out_of_bounds(rec)
-        strax.baseline(rec, baseline_samples=self.config['baseline_samples_aqmon'], flip=True)
-        aqmon_hits = strax.find_hits(rec, min_amplitude=self.config['hit_min_amplitude_aqmon'])
+        allowed_channels = [channel for hit_and_channel_list
+                            in self.hit_min_amplitude_aqmon for
+                            channel in hit_and_channel_list[1]]
+
+        not_allowed_channels = (set(np.unique(raw_records_aqmon['channel']))
+                                - set(allowed_channels))
+        if not_allowed_channels:
+            raise ValueError(f'Unknown channel {not_allowed_channels} {allowed_channels}')
+        records = strax.raw_to_records(raw_records_aqmon)
+        strax.sort_by_time(records)
+        strax.zero_out_of_bounds(records)
+        strax.baseline(records, baseline_samples=self.baseline_samples_aqmon, flip=True)
+        aqmon_hits = self.find_aqmon_hits_per_channel(records)
         return aqmon_hits
 
+    def find_aqmon_hits_per_channel(self, records):
+        """Allow different thresholds to be applied to different channels"""
+        aqmon_hits = [
+            strax.find_hits(
+                records[np.in1d(records['channel'], channels)],
+                min_amplitude=hit_threshold
+            )
+            for hit_threshold, channels in self.hit_min_amplitude_aqmon
+            if hit_threshold
+        ]
```

Comment:
This looks fairly complicated. Why are you not just passing the different thresholds as an array like we do for the TPC?

Reply:
good catch forgot about that posibility